main-board/src/sd_interface.rs

```diff
@@ -0,0 +1,152 @@
+
```

Comment:
Some general comments:
- Try using `rustfmt`. VSCode should have an integration where it runs it when saving. This will make this file more readable.
- Try to use imports instead of a full type name. E.g. `Timestamp` over `sd::Timestamp`. Again, this will improve readability.

Reply:
I agree with rustfmt and I've left some namespaces as it helps discern from two common imports. For example, pac::SERCOMx and hal::sercom::Sercomx are an example of ones that gave me issues.